### PR TITLE
Add doco for customized outbox isolation level

### PR DESCRIPTION
**Pull request by SzymonPobiega**

### CODE REVIEW

File: Snippets/SqlPersistence/SqlPersistence_6/OutboxSettings.cs [hunk 1]
```diff
@@ -44,6 +45,19 @@ void OutboxSettingsEx4(EndpointConfiguration endpointConfiguration)
         var outboxSettings = endpointConfiguration.EnableOutbox();
 
         outboxSettings.UseTransactionScope();
+        outboxSettings.UseTransactionScope(IsolationLevel.RepeatableRead);
```
Comment (mikeminutillo) [inline Snippets/SqlPersistence/SqlPersistence_6/OutboxSettings.cs:48]: Does this mean both lines will appear in the snippet? Should there be an `// OR` in there or something?
```suggestion
        // OR
        outboxSettings.UseTransactionScope(IsolationLevel.RepeatableRead);
```

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md [hunk 2]
```diff
@@ -36,4 +36,10 @@ In this mode the SQL persistence creates a `TransactionScope` that wraps the who
...
\ No newline at end of file
+The `TransactionScope` mode is most useful in legacy scenarios e.g. when migrating from MSMQ transport to a messaging infrastructure that does not support MS DTC. In order to maintain consistency the outbox has to be used in place of distributed transport-database transactions. If the legacy database cannot be modified to add the outbox table, the only option is to place the outbox table in a separate database and use distributed transactions between the databases.
+
+When necessary, the transaction isolation level can also be adjusted. This should be only done if the business logic that is executed by the handler within the outbox transaction requires higher isolation level to guarantee correctness (e.g. `RepeatableRead` or `Serializable`). Adjusting the isolation level is also possible when not using `TransactionScope` mode:
+
+snippet: SqlPersistenceOutboxIsolationLevel
+
+Note: Adjusting the isolation level requires version 6.1 of SQL Persistence.
```
Comment (mikeminutillo) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:45]: Could this section be a separate `[6.1,)` snippet?

```suggestion
Note: Adjusting the isolation level requires version 6.1 of SQL Persistence or higher.
```

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md [hunk 3]
```diff
@@ -36,4 +36,10 @@ In this mode the SQL persistence creates a `TransactionScope` that wraps the who
...
\ No newline at end of file
+The `TransactionScope` mode is most useful in legacy scenarios e.g. when migrating from MSMQ transport to a messaging infrastructure that does not support MS DTC. In order to maintain consistency the outbox has to be used in place of distributed transport-database transactions. If the legacy database cannot be modified to add the outbox table, the only option is to place the outbox table in a separate database and use distributed transactions between the databases.
+
+When necessary, the transaction isolation level can also be adjusted. This should be only done if the business logic that is executed by the handler within the outbox transaction requires higher isolation level to guarantee correctness (e.g. `RepeatableRead` or `Serializable`). Adjusting the isolation level is also possible when not using `TransactionScope` mode:
```
Comment (mikeminutillo) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:41]: Does this change the isolation level of the whole outbox transaction? What is the default?

Comment (SzymonPobiega) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:41]:
```suggestion
When necessary, the outbox transaction isolation level can also be adjusted. The change affects all data access done within this transaction. This should be only done if the business logic that is executed by the handler within the outbox transaction requires higher than default (`Read Committed`) isolation level to guarantee correctness (e.g. `Repeatable Read` or `Serializable`). Adjusting the isolation level is also possible when not using `TransactionScope` mode:
```

**Comment by SzymonPobiega** [comment]
Thanks @mikeminutillo . I reworded that paragraph a bit to make it more clear.

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md [hunk 4]
```diff
@@ -36,4 +36,10 @@ In this mode the SQL persistence creates a `TransactionScope` that wraps the who
...
 
-The `TransactionScope` mode is most useful in legacy scenarios e.g. when migrating from MSMQ transport to a messaging infrastructure that does not support MS DTC. In order to maintain consistency the outbox has to be used in place of distributed transport-database transactions. If the legacy database cannot be modified to add the outbox table, the only option is to place the outbox table in a separate database and use distributed transactions between the databases.
\ No newline at end of file
+The `TransactionScope` mode is most useful in legacy scenarios e.g. when migrating from MSMQ transport to a messaging infrastructure that does not support MS DTC. In order to maintain consistency the outbox has to be used in place of distributed transport-database transactions. If the legacy database cannot be modified to add the outbox table, the only option is to place the outbox table in a separate database and use distributed transactions between the databases.
```
Comment (HEskandari) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:39]:
```suggestion
The `TransactionScope` mode is most useful in legacy scenarios such as when migrating from MSMQ transport to a messaging infrastructure that does not support MSDTC. In order to maintain consistency, the outbox has to be used in place of distributed transport-database transactions. If the legacy database cannot be modified to add the outbox table, the only option is to place the outbox table in a separate database and use distributed transactions between the databases.
```

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md [hunk 5]
```diff
@@ -36,4 +36,10 @@ In this mode the SQL persistence creates a `TransactionScope` that wraps the who
...
+
+When necessary, the outbox transaction isolation level can also be adjusted. The change affects all data access done within this transaction. This should be only done if the business logic that is executed by the handler within the outbox transaction requires higher than default (`Read Committed`) isolation level to guarantee correctness (e.g. `Repeatable Read` or `Serializable`). Adjusting the isolation level is also possible when not using `TransactionScope` mode:
```
Comment (HEskandari) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:41]:
```suggestion
When necessary, the outbox transaction isolation level can also be adjusted. The change in the isolation level affects all data access operations done within the transaction. This should be only done if the business logic that is executed by the handler within the outbox transaction requires higher than default (`Read Committed`) isolation level to guarantee correctness (e.g. `Repeatable Read` or `Serializable`). Adjusting the isolation level is also possible when not using `TransactionScope` mode:
```

**Review by HEskandari — COMMENTED** [review]
Made small suggestions, but looks good.

**Pull request merged**